spec/Functions.tex

```diff
@@ -767,6 +764,43 @@ \subsection{The Ref Return Intent}
 \end{chapeloutput}
 \end{chapelexample}
 
+
+\subsection{The Const Ref Return Intent}
+\label{Const_Ref_Return_Intent}
+\index{functions!const ref keyword and@\chpl{const ref} keyword and}
+\index{const ref (return intent)@\chpl{const ref} (return intent)}
+\index{intents!const ref return@\chpl{const ref} return}
+
+The \chpl{const ref} return intent is also available. It is a restricted
+form of the \chpl{ref} return intent. In particular, it uses a similar
+mechanism but the function call cannot be assigned to. The \chpl{const ref}
+return intent should be viewed as a performance optimization. It
+indicates that the returned value does not need to be copied because the
+returned value will not be destroyed when the function returns.
+
```

Comment:
I am having a hard time with this in the context of a spec.

a) I don't think there should be any mention of the "mechanism" involved in a return.

b) I am also leary of describing a language feature "as a" performance optimization.  Performance
optimization is a quality of a number of complicated features.  It's true that language may have an
impact on the ability of the complier/runtime to optimize for performance but it's a not a quality of
a language feature.

c) Combining (a) and (b), I think that it's a mistake to talk about "copying" return values.  It's the
wrong discussion for a spec and it might create confusion about return with ref-intent. It is not
necessarily the case that a function that returns with ref-intent will do any copying either.  I think that
copying only occurs if the lvalue generated by the function is assigned to an rvalue.